java-checks/src/main/java/org/sonar/java/checks/spring/MissingPathVariableAnnotationCheck.java

```diff
@@ -0,0 +1,374 @@
+    if (requestMappingArguments != null) {
+      Set<String> templateVars;
+      try {
+        templateVars = templateVariablesFromMapping(requestMappingArguments);
+      } catch (DoNotReport ignored) {
+        return;
+      }
+
+      if (templateVars != null) {
+        requestMappingTemplateVariables = templateVars;
+      }
+    }
+
+    // we find path variable annotations on method annotated with @ModelAttribute and extract the name
+    Set<String> modelAttributeMethodParameter = new HashSet<>();
+    for (var method : methods) {
+      if (!method.symbol().metadata().isAnnotatedWith(MODEL_ATTRIBUTE_ANNOTATION)) {
+        continue;
+      }
+      for (var parameter : method.parameters()) {
+        SymbolMetadata metadata = parameter.symbol().metadata();
+        var arguments = metadata.valuesForAnnotation(PATH_VARIABLE_ANNOTATION);
+        if (arguments != null) {
+          modelAttributeMethodParameter.add(extractPathMethodParameters(parameter, arguments).value);
+        }
+      }
+    }
+
+    for (var method : methods) {
+      if (!method.symbol().metadata().isAnnotatedWith(MODEL_ATTRIBUTE_ANNOTATION)) {
+        try {
+          checkParametersAndPathTemplate(method, modelAttributeMethodParameter, requestMappingTemplateVariables);
+        } catch (DoNotReport ignored) {
+        }
+      }
+    }
+  }
+
+  private void checkParametersAndPathTemplate(MethodTree method, Set<String> modelAttributeMethodParameters, Set<String> requestMappingTemplateVars) {
+    // we find path variable annotations and extract the name
+    // example find : @PathVariable() String id and extract id
+    List<ParameterInfo> methodParameters = new ArrayList<>();
+    for (var parameter : method.parameters()) {
+      SymbolMetadata metadata = parameter.symbol().metadata();
+
+      if (metadata.annotations().stream().anyMatch(ann -> ann.symbol().isUnknown())) {
+        throw new DoNotReport();
+      }
+
+      var arguments = metadata.valuesForAnnotation(PATH_VARIABLE_ANNOTATION);
+      if (arguments != null) {
+        methodParameters.add(extractPathMethodParameters(parameter, arguments));
+      }
+
+    }
+
+    // we find mapping annotation and extract path
+    // example find @GetMapping("/{id}") and extract "/{id}"
+    List<UriInfo<Set<String>>> templateVariables = new ArrayList<>();
+    for (var ann : method.modifiers().annotations()) {
+      if (ann.symbolType().isUnknown()) {
+        throw new DoNotReport();
+      }
+
+      String fullyQualifiedName = ann.annotationType().symbolType().fullyQualifiedName();
+      var values = method.symbol().metadata().valuesForAnnotation(fullyQualifiedName);
+      if (values == null || !MAPPING_ANNOTATIONS.contains(fullyQualifiedName)) {
+        continue;
+      }
+
+      @Nullable
+      var templatesVars = templateVariablesFromMapping(values);
+
+      if (templatesVars != null) {
+        templateVariables.add(new UriInfo<>(ann, templatesVars));
+      }
```

Comment:
If we change the `templateVariablesFromMapping` method we can also simplify here
```suggestion
    templateVariables.add(new UriInfo<>(ann, templateVariablesFromMapping(values)));
```

Reply:
:+1: